pkg/updater/updater_test.go

```diff
@@ -0,0 +1,165 @@
+					{ Name: "test-rack-1", Members: 1 },
+					{ Name: "test-rack-2", Members: 2 },
+					{ Name: "test-rack-3", Members: 3 },
+					{ Name: "test-rack-4", Members: 1 },
+				},
+			},
+		},
+		Status: scyllav1.ClusterStatus{
+			Racks: map[string]scyllav1.RackStatus{
+				"test-rack-1": { Members: 1, ReadyMembers: 1 },
+				"test-rack-2": { Members: 2, ReadyMembers: 1 },
+				"test-rack-3": { Members: 2, ReadyMembers: 2 },
+				"test-rack-4": { Members: 1, ReadyMembers: 1 },
+			},
+		},
+	}
+
+	autoUpdateMode := v1alpha1.UpdateModeAuto
+	modeAutoSca := v1alpha1.ScyllaClusterAutoscaler{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-auto-sca",
+			Namespace: "test-sca-ns",
+		},
+		Spec: v1alpha1.ScyllaClusterAutoscalerSpec{
+			TargetRef: &v1alpha1.TargetRef{
+				Namespace: "test-cluster-ns",
+				Name:      "test-cluster",
+			},
+			UpdatePolicy: &v1alpha1.UpdatePolicy{
+				UpdateMode: &autoUpdateMode,
+			},
+		},
+		Status: v1alpha1.ScyllaClusterAutoscalerStatus{
+			Recommendations: &v1alpha1.ScyllaClusterRecommendations{
+				DataCenterRecommendations: []v1alpha1.DataCenterRecommendations{
+					{
+						Name:                "test-dc",
+						RackRecommendations: []v1alpha1.RackRecommendations{
+							{ Name: "test-rack-1", Members: &v1alpha1.RecommendedRackMembers{ Target: 2 } },
+							{ Name: "test-rack-2", Members: &v1alpha1.RecommendedRackMembers{ Target: 3 } },
+							{ Name: "test-rack-3", Members: &v1alpha1.RecommendedRackMembers{ Target: 2 } },
+						},
+					},
+				},
+			},
+		},
+	}
+
+	offUpdateMode := v1alpha1.UpdateModeOff
+	modeOffSca := v1alpha1.ScyllaClusterAutoscaler{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-off-sca",
+			Namespace: "test-sca-ns",
+		},
+		Spec: v1alpha1.ScyllaClusterAutoscalerSpec{
+			TargetRef: &v1alpha1.TargetRef{
+				Namespace: "test-cluster-ns",
+				Name:      "test-cluster",
+			},
+			UpdatePolicy: &v1alpha1.UpdatePolicy{
+				UpdateMode: &offUpdateMode,
+			},
+		},
+		Status: v1alpha1.ScyllaClusterAutoscalerStatus{
+			Recommendations: &v1alpha1.ScyllaClusterRecommendations{
+				DataCenterRecommendations: []v1alpha1.DataCenterRecommendations{
+					{
+						Name:                "test-dc",
+						RackRecommendations: []v1alpha1.RackRecommendations{
+							{ Name: "test-rack-4", Members: &v1alpha1.RecommendedRackMembers{ Target: 2 } },
+						},
+					},
+				},
+			},
+		},
+	}
+
+	clientBuilder := fake.NewClientBuilder().WithScheme(scheme).
+		WithObjects(&testScyllaCluster, &modeAutoSca, &modeOffSca)
+	c := clientBuilder.Build()
+	atom := zap.NewAtomicLevelAt(zapcore.DebugLevel)
+	logger, _ := log.NewProduction(log.Config{Level: atom})
+	u := NewUpdater(c, logger)
+
+	ctx := context.Background()
+	err := u.RunOnce(ctx)
+	require.NoError(t, err, "Updater RunOnce. Message: '%s'", err)
+
+	tests := []struct{
+		name string
+		rackName string
+		expectedMembersValue int32
+	}{
+		{
+			name: "applied recommendation",
+			rackName: "test-rack-1",
```

Comment:
It would be much clearer if test input would be pair of recommendation and cluster, and expected state. Right now just by looking at the test input, it's not clear what it verfies.
It is also hard to extend it, because you're running updater outside of individual test.